sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -296,32 +299,42 @@ private Optional<SclReportItem> updateSourceDataSetsAndControlBlocks(TExtRef ext
             return warningReportItem(extRef, String.format(MESSAGE_SOURCE_LN_NOT_FOUND, optionalSourceLDevice.get().getXPath()));
         }
 
-        //TODO: Don't forget to test this in issue #84 RSR-433
-        List<ResumedDataTemplate> mxAndStSources = sourceDas.stream()
-            .filter(da -> da.getFc() == TFCEnum.MX || da.getFc() == TFCEnum.ST).toList();
+        return filterValidSourceDas(extRef, sourceDas);
+    }
 
-        Stream<ResumedDataTemplate> sourceDaToProcess;
+    private Optional<SclReportItem> filterValidSourceDas(TExtRef extRef, final Set<ResumedDataTemplate> sourceDas) {
+        //TODO: Don't forget to test this in issue #84 RSR-433
+        sourceDas.removeIf(da -> da.getFc() != TFCEnum.MX && da.getFc() != TFCEnum.ST);
         switch (extRef.getServiceType()) {
-            case GOOSE:
-            case SMV:
-                sourceDaToProcess = mxAndStSources.stream().filter(FcdaCandidates.SINGLETON::contains);
-                break;
-            case REPORT:
-                //TODO: In issue RSR-608, replace with mxAndStSources.filter(...)
-                sourceDaToProcess = Stream.empty();
-                break;
-            case POLL:
-            default:
+            case GOOSE, SMV -> sourceDas.removeIf(Predicate.not(FcdaCandidates.SINGLETON::contains));
+            case REPORT -> {
+                Optional<SclReportItem> filteringError = filterReportSourceDa(extRef, sourceDas);
+                if (filteringError.isPresent()) {
+                    return filteringError;
+                }
+            }
+            default -> {
                 return fatalReportItem(extRef, String.format(MESSAGE_INVALID_SERVICE_TYPE, extRef.getServiceType()));
+            }
         }
         //TODO: map to FCDA in issue #84 RSR-433 and remove print to console
-        String daToPrint = sourceDaToProcess
+        String daToPrint = sourceDas.stream()
             .map(da -> da.getFc() + "#" + da.getDataAttributes())
             .collect(Collectors.joining(","));
         System.out.print("," + (daToPrint.isEmpty() ? "--NO_VALID_SOURCE_DA--" : daToPrint));
         return Optional.empty();
     }
 
+    private Optional<SclReportItem> filterReportSourceDa(TExtRef extRef, Set<ResumedDataTemplate> sourceDas) {
+        String[] extRefDescParts = extRef.getDesc().split(EXTREF_DESC_DELIMITER);
```

Comment:
`filterReportSourceDa` need to be tested separately, so it can be no private.

Reply:
OK. I extracted a method in Utils to extract the nth fiield starting from the end.